cli/client.go

```diff
@@ -380,6 +387,86 @@ func RobotsPartShellAction(c *cli.Context) error {
 	)
 }
 
+// checkUpdateResponse holds the values used to hold release information.
+type checkUpdateResponse struct {
+	Name       string `json:"name"`
+	TagName    string `json:"tag_name"`
+	TarballURL string `json:"tarball_url"`
+}
+
+// CheckUpdateAction is the corresponding Action for 'check-update'.
+func CheckUpdateAction(c *cli.Context) error {
+	if c.Bool(quietFlag) {
+		return nil
+	}
+
+	dateCompiledRaw := rconfig.DateCompiled
+	dateCompiled, err := time.Parse("2006-01-02", dateCompiledRaw)
+	if err != nil {
+		warningf(c.App.ErrWriter, "failed to parse compiled date: %w", err)
+		return nil
+	}
+
+	// compare to current version
+	appVersion := rconfig.Version
+
+	// skip check if stable & install is older than two week
+	if appVersion != "" && time.Since(dateCompiled) < time.Hour*24*7*2 {
+		return nil
+	}
+
+	// skip check if on dev/latest & install and less than 6wks old
+	if appVersion == "" && time.Since(dateCompiled) < time.Hour*24*7*6 {
+		return nil
+	}
+
+	v1, err := semver.NewVersion(appVersion)
+	if err != nil {
+		warningf(c.App.ErrWriter, "failed to parse compiled version: %w", err)
+		return nil
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Millisecond*80)
+	defer cancel()
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, releaseURL, nil)
+	if err != nil {
+		warningf(c.App.ErrWriter, "failed to create request: %w", err)
+		return nil
+	}
+
+	client := http.DefaultClient
+	res, err := client.Do(req)
+	if err != nil {
+		warningf(c.App.ErrWriter, "failed to get latest release information: %w", err)
+		return nil
+	}
+
+	resp := checkUpdateResponse{}
+	err = json.NewDecoder(res.Body).Decode(&resp)
+	if err != nil {
+		warningf(c.App.ErrWriter, "failed to decode response: %w", err)
+		return nil
+	}
+	err = res.Body.Close()
```

Comment:
[opt/style] We generally prefer to close with a defer. Not sure if there is a reason not to here.

Reply:
I kept failing the `build-go lint-go` when using defer for anything

Reply:
```
cli/client.go:451:22: Error return value of `res.Body.Close` is not checked (errcheck)
        defer res.Body.Close()
```

Reply:
You should put the defer above the json parsing

Reply:
You should put this in a Utils.UncheckedErrorFunc (and also above json parsing)